Modules/events/event.py

```diff
@@ -0,0 +1,181 @@
+"""
+event.py - Event system
+
+A Class decorator event system.
+
+Copyright (c) 2018 The Fuel Rats Mischief,
+All rights reserved.
+
+Licensed under the BSD 3-Clause License.
+
+See LICENSE.md
+"""
+import logging
+from functools import partial
+from typing import Callable, List, Dict, Any, Union
+
+# typedef
+Subscriptions = List[Callable]
+log = logging.getLogger(f"mecha.{__name__}")
+
+# If a subscriber returns this value, the event will canceled.
+# return this value to consume an event
+CANCEL_EVENT = object()
```

Comment:
Could you make it clear what 'cancelling' an event means? I assume it prevents other subscribers from being called, but it isn't clear. Also, it would be great if this info were added to `Event.subscribe`.

Reply:
sure thing! il add further documentation and revise what exists.